### ISSUE TITLE
OneStatementPerLine: multiple empty statements not reported despite documentation

### ISSUE THREAD

**Issue by VeniceWish** [checkstyle/checkstyle#17662]
I have read check documentation: https://checkstyle.sourceforge.io/checks/coding/onestatementperline.html
I have downloaded the latest checkstyle from: https://checkstyle.org/cmdline.html#Download_and_Run
I have executed the cli and showed it below, as cli describes the problem better than 1,000 words

Code:
```bash
public class Test { 
  void testEmptyStatement() {
    ; ; // Expected Violation
  }

  void testNonEmptyStatement() {
    int x = 0; int y = 1; int z = 1; // 2 Violations reported
  }

  void testMixed() {
    ; ; int x = 0; // Expected Violation
    ; int y = 0; ; // Expected Violation
    int z = 0;;;   // Expected Violation
  }
}
```
Config:
```
/var/tmp $ cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Checkstyle//DTD Checkstyle Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
    <property name="localeLanguage" value="en"/>
    <module name="TreeWalker">
        <module name="OneStatementPerLine"/>
    </module>
</module>
```
Cli:
```
/var/tmp $ java -jar checkstyle-10.26.1-all.jar -c config.xml Test.java
Starting audit...
[ERROR] Test.java:7:25: Only one statement per line allowed. [OneStatementPerLine]
[ERROR] Test.java:7:36: Only one statement per line allowed. [OneStatementPerLine]
Audit done.
Checkstyle ends with 2 errors.
```

---

According to the OneStatementPerLine documentation:

> OneStatementPerLineCheck checks the following types of statements: variable declaration statements, empty statements, import statements, assignment statements, expression statements, increment statements, object creation statements, 'for loop' statements, 'break' statements, 'continue' statements, 'return' statements, resources statements (optional).

This clearly indicates that empty statements should be reported.

However, in the provided example, multiple empty statements  are not reported as violations. Only multiple non-empty statements are flagged.

This suggests an inconsistency between the documentation and the actual implementation of the OneStatementPerLine check.

---

**Comment by romani** [comment]
We need to recheck implementation.
All logic is at https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/OneStatementPerLineCheck.java

There was something, some nuance, on such extra `;`.
Do you asking in scope of Google style or not?

Either implementation or doc needs to be updated.

**Comment by VeniceWish** [comment]
Thanks for the reply. Yes, my question is in scope of Google Style.
However, Google Style only states one short rule: 
> “Each statement is followed by a line break.”

To better understand how Checkstyle interprets this, I looked into the documentation of `OneStatementPerLine`. The documentation explicitly says that empty statements are also checked. But when I tested it, empty statements were not reported as violations, so I noticed an inconsistency between the documentation and the actual implementation.

I believe your interpretation of the Google Style rule is correct, and it should include empty statements as well. So perhaps it is the implementation that needs to be adjusted.

**Comment by romani** [comment]
Issue is approved

**Comment by romani** [comment]
@Tieshiyang , can you share behavior of Google formatter?

**Comment by VeniceWish** [comment]
Formatter:
```
$ java -jar google-java-format-1.28.0-all-deps.jar Test.java > FormattedCode.java
$ diff -Naru Test.java FormattedCode.java
--- Test.java   2025-08-25 16:49:26.528598700 +0800
+++ FormattedCode.java  2025-08-25 16:49:33.025840200 +0800
@@ -1,15 +1,24 @@
-public class Test {
+public class Test {
   void testEmptyStatement() {
-    ; ; // Expected Violation
+    ;
+    ; // Expected Violation
   }

   void testNonEmptyStatement() {
-    int x = 0; int y = 1; int z = 1; // Expected Violation
+    int x = 0;
+    int y = 1;
+    int z = 1; // Expected Violation
   }

   void testMixed() {
-    ; ; int x = 0; // Expected Violation
-    ; int y = 0; ; // Expected Violation
-    int z = 0;;;   // Expected Violation
+    ;
+    ;
+    int x = 0; // Expected Violation
+    ;
+    int y = 0;
+    ; // Expected Violation
+    int z = 0;
+    ;
+    ; // Expected Violation
   }
-}
\ No newline at end of file
+}
```

**Comment by romani** [comment]
Fix is merged
